src/Cache/Cache.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Cache;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+class Cache implements CacheInterface
+{
+    /** @var CacheItemPoolInterface */
+    private $cachePool;
+
+    /** @var string */
+    private $prefix;
+
+    /** @var bool */
+    private $flushCacheItems = false;
+
+    private $defaultCacheTimes = [
+        CacheInterface::NONE => -1,
+        CacheInterface::SHORT => 60,
+        CacheInterface::NORMAL => 300,
+        CacheInterface::MEDIUM => 1200,
+        CacheInterface::LONG => 7200,
+        CacheInterface::X_LONG => 86400,
+        CacheInterface::INDEFINITE => 0,
+    ];
+
+    /** @var array */
+    private $cacheTimes;
+
+    public function __construct(
+        CacheItemPoolInterface $cachePool,
+        string $prefix,
+        array $cacheTimes = []
+    ) {
+        $this->cachePool = $cachePool;
+        $this->prefix = $prefix;
+        $this->cacheTimes = array_merge($this->defaultCacheTimes, $cacheTimes);
+    }
+
+    public function getItem(string $key): CacheItemInterface
+    {
+        $key = $this->standardiseKey($key);
+        if ($this->flushCacheItems) {
+            $this->cachePool->deleteItem($key);
+        }
+        return $this->cachePool->getItem($key);
+    }
+
+    public function setItem(CacheItemInterface $item, $ttl): bool
```

Comment:
Is $ttl an int?  
Also, should there be a docblock to say that the ttl is in seconds? Or is that obvious enough?